### DISCUSSION TITLE
Building OVOS from source

### DISCUSSION THREAD

**Discussion by unfirthman** [OpenVoiceOS/ovos-buildroot#215] [Dev]
Hello, all. I would like to build OVOS from source for two main reasons: I don't want OVOS to automatically start on boot - I would like to ability to start / stop OVOS manually ; and I'm only planning on using some services / modules and don't require a big intensive setup.

I've been following instructions on the ovos-core readme.md and the instructions [here](https://openvoiceos.github.io/community-docs/042-install_ovos_core/) however, any time I use the command `pip install ovos-core[mycroft] ` my terminal returns that no matches are found for `ovos-core[mycroft]`

I am sure to be missing a step somewhere. Can anyone provide detailed instructions on how to install via source?

Thanks!

**Comment by ChanceNCounter** [comment]
I apologize for a fairly lazy drive-by, but I want to get you going quickly.

This is pretty young, but it’s meant to be the quickest “long way” if you aren’t actually after a *development* build. https://github.com/OpenVoiceOS/ovos-installer

If you really do want to build from source, what you’re missing is the `pre` flag in that pip command (my phone is absolutely refusing to type a double hyphen in a text field, but on this unusual occasion I can recommend copying and pasting the pip command from the docs)

If you decide to use the installer, which sounds like it should suit your use case, please don’t hesitate to speak up about any hiccups or difficulties. (don’t hesitate in either case, but I’d generally expect the installer to be a smoother ride than source, due to an ongoing, org-wide sprint to facilitate better packaging)

**Reply by unfirthman** [reply]
Thank you for this! And no apologies necessary - I appreciate the help. I did use the ovos-installer earlier today, and it did get the job done. However, I wasn't able to figure out how to exit OVOS or disable OVOS from automatically starting on boot. I tried digging around that installation a bit, then decided to try building from source thinking that would be a quicker way of understanding how to customize my config...

I think what I will do next is use the official ovos-installer, then come back to this board to ask questions and report back. Sounds like this would be the most mutually beneficial!

**Reply by goldyfruit** [reply]
Here is a quick loop to shutdown OVOS services.

```bash
sudo systemctl stop ovos-phal-admin
for svc in ovos-core ovos-listener ovos-messagebus ovos-phal; do
  systemctl --user stop $svc
done
```

**Reply by unfirthman** [reply]
Brilliant! Thank you for this

**Comment by unfirthman** [comment]
Thanks for the help, everyone. I used the installer to get a working build. Also, many of the executables I was looking for are in the .venvs/ovos/bin directory - which is quite helpful.

Maybe good for you all to know that OVOS works with the WM8960 audio hat once the drivers are installed. No trouble at all!

I did come across one easy documentation fix that I will document in a PR. Also, the exit loop is very helpful for toggling services. Is that something you would consider adding to ovos-core? I'm happy to put that in a PR as something like `ovos-exit`
